nixos/modules/services/monitoring/zabbix-proxy.nix

```diff
@@ -0,0 +1,133 @@
+# Zabbix proxy daemon.
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+
+  cfg = config.services.zabbixProxy;
+
+  stateDir = "/run/zabbix";
+
+  logDir = "/var/log/zabbix";
+
+  libDir = "/var/lib/zabbix";
+
+  pidFile = "${stateDir}/zabbix_proxy.pid";
+
+  configFile = pkgs.writeText "zabbix_proxy.conf"
+    ''
+      Server = ${cfg.server}
+
+      LogFile = ${logDir}/zabbix_proxy
+
+      PidFile = ${pidFile}
+
+      ${optionalString (cfg.dbServer != "localhost") ''
+        DBHost = ${cfg.dbServer}
+      ''}
+
+      DBName = zabbix
+
+      DBUser = zabbix
+
+      ${optionalString (cfg.dbPassword != "") ''
+        DBPassword = ${cfg.dbPassword}
+      ''}
+
+      ${config.services.zabbixProxy.extraConfig}
+    '';
+
+  useLocalPostgres = cfg.dbServer == "localhost" || cfg.dbServer == "";
+
+in
+
+{
+
+  ###### interface
+
+  options = {
```

Comment:
Less typing preferred, instead of writing `services.zabbixProxy.*` over and over you can write `services.zabbixProxy = { ... };`